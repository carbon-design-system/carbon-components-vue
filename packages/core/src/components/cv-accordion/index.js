import CvAccordion from './cv-accordion';
import CvAccordionItem from './cv-accordion-item';
import CvAccordionSkeleton from './cv-accordion-skeleton';

export { CvAccordion, CvAccordionItem, CvAccordionSkeleton };
