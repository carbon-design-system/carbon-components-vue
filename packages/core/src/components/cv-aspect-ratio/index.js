import CvAspectRatio from './cv-aspect-ratio';
export { CvAspectRatio };
