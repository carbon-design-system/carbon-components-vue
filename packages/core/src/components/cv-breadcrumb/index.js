import CvBreadcrumb from './cv-breadcrumb';
import CvBreadcrumbItem from './cv-breadcrumb-item';
import CvBreadcrumbSkeleton from './cv-breadcrumb-skeleton';

export { CvBreadcrumb, CvBreadcrumbItem, CvBreadcrumbSkeleton };
