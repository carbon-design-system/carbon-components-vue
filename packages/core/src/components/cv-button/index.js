import CvButton from './cv-button';
import CvIconButton from './cv-icon-button';
import CvButtonSkeleton from './cv-button-skeleton';
import CvButtonSet from './cv-button-set';

export { CvButton, CvIconButton, CvButtonSkeleton, CvButtonSet };
