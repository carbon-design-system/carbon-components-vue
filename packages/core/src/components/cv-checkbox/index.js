import CvCheckbox from './cv-checkbox';
import CvCheckboxSkeleton from './cv-checkbox-skeleton';
export { CvCheckbox, CvCheckboxSkeleton };
