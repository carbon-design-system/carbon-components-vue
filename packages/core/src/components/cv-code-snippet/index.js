import CvCodeSnippet from './cv-code-snippet';
import CvCodeSnippetSkeleton from './cv-code-snippet-skeleton';
export { CvCodeSnippet, CvCodeSnippetSkeleton };
