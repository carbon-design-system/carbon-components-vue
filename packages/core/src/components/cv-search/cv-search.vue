<template>
  <cv-wrapper :tag-type="formItem ? 'div' : ''" :class="`cv-search ${carbonPrefix}--form-item`">
    <div
      :class="[
        `${carbonPrefix}--search`,
        {
          [`${carbonPrefix}--search--${theme}`]: theme.length,
          [`${carbonPrefix}--search--${internalSize}`]: internalSize,
          [`${carbonPrefix}--toolbar-search`]: isToolbarKind,
          [`${carbonPrefix}--toolbar-search--active`]: toolbarActive,
          'cv-search': !formItem,
        },
      ]"
      role="search"
      ref="search"
    >
      <label :for="uid" :class="`${carbonPrefix}--label`">{{ label }}</label>

      <input
        :id="uid"
        :class="`${carbonPrefix}--search-input`"
        v-bind="$attrs"
        v-model="internalValue"
        v-on="inputListeners"
        type="text"
        role="search"
        ref="input"
        :placeholder="placeholder"
        :aria-labelledby="uid"
        @blur="checkFocus"
      />

      <button
        type="button"
        v-if="isToolbarKind"
        :class="`${carbonPrefix}--toolbar-search__btn`"
        :aria-label="toolbarAriaLabel"
        @click="toggleActive(true)"
        @blur="checkFocus"
      >
        <component :is="icon" :class="`${carbonPrefix}--search-magnifier`" />
      </button>
      <component v-if="!isToolbarKind" :is="icon" :class="`${carbonPrefix}--search-magnifier`" />
      <button
        type="button"
        :class="[`${carbonPrefix}--search-close`, { [`${carbonPrefix}--search-close--hidden`]: !clearVisible }]"
        :title="clearAriaLabel"
        :aria-label="clearAriaLabel"
        @click="onClearClick"
      >
        <Close16 />
      </button>
    </div>
  </cv-wrapper>
</template>

<script>
import { uidMixin, themeMixin, carbonPrefixMixin, methodsMixin } from '../../mixins';
import Search16 from '@carbon/icons-vue/es/search/16';
import Search20 from '@carbon/icons-vue/es/search/20';
import Close16 from '@carbon/icons-vue/es/close/16';
import CvWrapper from '../cv-wrapper/_cv-wrapper';

export default {
  name: 'CvSearch',
  mixins: [uidMixin, themeMixin, carbonPrefixMixin, methodsMixin({ input: ['blur', 'focus'] })],
  components: { Close16, CvWrapper },
  inheritAttrs: false,
  props: {
    clearAriaLabel: { type: String, default: 'Clear search input' },
    formItem: { type: Boolean, default: true },
    kind: { type: String, default: undefined },
    label: String,
    size: { type: String, default: undefined },
    small: {
      type: Boolean,
      default: undefined,
      validator(val) {
        if (val !== undefined && process.env.NODE_ENV === 'development') {
          console.warn('DEPRECARTED: Prefer size property: small, large or xl (default)');
        }
        return true;
      },
    },
    large: {
      type: Boolean,
      default: undefined,
      validator(val) {
        if (val !== undefined && process.env.NODE_ENV === 'development') {
          console.warn('DEPRECARTED: Prefer size property: small, large or xl (default)');
        }
        return true;
      },
    },
    value: String,
    placeholder: { type: String, default: 'Search' },
    toolbarAriaLabel: { type: String, default: 'Toolbar search' },
  },
  data() {
    return {
      clearVisible: this.value ? this.value.length : false,
      internalValue: this.value,
      toolbarActive: false,
    };
  },
  watch: {
    value() {
      this.clearVisible = this.value ? this.value.length : false;
      this.internalValue = this.value;
    },
  },
  computed: {
    // Bind listeners at the component level to the embedded input element and
    // add our own input listener to service the v-model. See:
    // https://vuejs.org/v2/guide/components-custom-events.html#Customizing-Component-v-model
    inputListeners() {
      return {
        ...this.$listeners,
        input: this.onInput,
      };
    },
    internalSize() {
      let size;

      if (this.size !== undefined && (this.size || (this.small === undefined && this.large === undefined))) {
        switch (this.size) {
          case 'small':
            size = 'sm';
            break;
          case 'large':
            size = 'lg';
            break;
          default:
            size = 'xl';
            break;
        }
      } else {
        size = this.small ? 'sm' : 'xl';
      }
      return size;
    },
    isToolbarKind() {
      return this.kind === 'toolbar';
    },
    icon() {
      return this.size === 'xl' ? Search20 : Search16;
    },
  },
  methods: {
    onClearClick() {
      this.internalValue = '';
      this.clearVisible = false;
      this.$emit('input', this.internalValue);
    },
    onInput() {
      this.clearVisible = this.internalValue.length > 0;
      return this.$emit('input', this.internalValue);
    },
    toggleActive(force) {
      if (typeof force === 'boolean') {
        this.toolbarActive = force;
      } else {
        this.toolbarActive = !this.toolbarActive;
      }
      if (this.toolbarActive) {
        this.$refs.input.focus();
      }
    },
    checkFocus(ev) {
      if (this.isToolbarKind) {
        if (!this.$el.contains(ev.relatedTarget)) {
          this.toolbarActive = false;
        }
      }
    },
  },
};
</script>
