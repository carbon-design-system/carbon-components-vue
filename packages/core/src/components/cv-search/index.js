import CvSearch from './cv-search';
export { CvSearch };
