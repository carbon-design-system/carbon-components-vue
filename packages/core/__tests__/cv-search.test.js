import { testComponent, awaitNextTick } from './_helpers';
const { shallowMount: shallow, mount, trigger, setProps } = awaitNextTick;
import { CvSearch } from '@/components/cv-search';
import Search16 from '@carbon/icons-vue/es/search/16';
import Search20 from '@carbon/icons-vue/es/search/20';

describe('CvSearch', () => {
  // ***************
  // PROP CHECKS
  // ***************
  testComponent.propsAreType(
    CvSearch,
    ['clearAriaLabel', 'kind', 'label', 'size', 'value', 'placeholder', 'toolbarAriaLabel'],
    String
  );
  testComponent.propsAreType(CvSearch, ['formItem', 'small', 'large'], Boolean);

  testComponent.propsHaveDefault(CvSearch, ['clearAriaLabel', 'formItem', 'placeholder', 'toolbarAriaLabel']);
  testComponent.propsHaveDefaultOfUndefined(CvSearch, ['kind', 'size', 'small', 'large']);

  // ***************
  // SNAPSHOT TESTS
  // ***************
  it('should render correctly with only `value` prop set', async () => {
    const propsData = { id: 'test-1', value: 'check-1' };
    const wrapper = await mount(CvSearch, { propsData });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render correctly when `size` is small', async () => {
    const propsData = { id: 'test-1', value: 'check-1', size: 'small' };
    const wrapper = await shallow(CvSearch, { propsData });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render correctly when `size` is large', async () => {
    const propsData = { id: 'test-1', value: 'check-1', size: 'large' };
    const wrapper = await shallow(CvSearch, { propsData });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render correctly when `size` is xl', async () => {
    const propsData = { id: 'test-1', value: 'check-1', size: 'xl' };
    const wrapper = await shallow(CvSearch, { propsData });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render correctly when `clearAriaLabel` is set', async () => {
    const propsData = { id: 'test-1', value: 'check-1', clearAriaLabel: 'test clear aria label' };
    const wrapper = await shallow(CvSearch, { propsData });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render correctly when `formItem` is false', async () => {
    const propsData = { id: 'test-1', value: 'check-1', formItem: false };
    const wrapper = await shallow(CvSearch, { propsData });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render correctly when `kind` is toolbar', async () => {
    const propsData = { id: 'test-1', value: 'check-1', kind: 'toolbar' };
    const wrapper = await shallow(CvSearch, { propsData });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render correctly when `placeholder` is set', async () => {
    const propsData = { id: 'test-1', value: 'check-1', placeholder: 'test placeholder' };
    const wrapper = await shallow(CvSearch, { propsData });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render correctly when `toolbarAriaLabel` is set', async () => {
    const propsData = { id: 'test-1', value: 'check-1', toolbarAriaLabel: 'test toolbar aria label', kind: 'toolbar' };
    const wrapper = await shallow(CvSearch, { propsData });

    expect(wrapper.html()).toMatchSnapshot();
  });

  // ***************
  // FUNCTIONAL TESTS
  // ***************

  it('`clearVisible` is computed correctly', async () => {
    const value = 'check-1';
    const propsData = { id: 'test-1', value };
    const wrapper = await shallow(CvSearch, { propsData });

    expect(wrapper.vm.clearVisible).toEqual(value.length);
  });

  it('`internalValue` is computed correctly', async () => {
    const value = 'check-1';
    const propsData = { id: 'test-1', value };
    const wrapper = await shallow(CvSearch, { propsData });

    expect(wrapper.vm.internalValue).toEqual(value);
  });

  it('`toolbarActive` is initially set correctly', async () => {
    const value = 'check-1';
    const propsData = { id: 'test-1', value };
    const wrapper = await shallow(CvSearch, { propsData });

    expect(wrapper.vm.toolbarActive).toBeFalsy();
  });

  it('`value` prop should be watched', async () => {
    const propsData = { value: 'test value', id: '1' };
    const wrapper = await shallow(CvSearch, { propsData });

    const newValue = 'qwerty';
    await setProps(wrapper, {
      value: newValue,
    });
    expect(wrapper.vm.internalValue).toEqual(newValue);
    expect(wrapper.vm.clearVisible).toEqual(newValue.length);
  });

  it('`isToolbarKind` is computed correctly, falsy', async () => {
    const value = 'check-1';
    const propsData = { id: 'test-1', value, kind: 'not toolbar' };
    const wrapper = await shallow(CvSearch, { propsData });

    expect(wrapper.vm.isToolbarKind).toBeFalsy();
  });

  it('`isToolbarKind` is computed correctly, truthy', async () => {
    const value = 'check-1';
    const propsData = { id: 'test-1', value, kind: 'toolbar' };
    const wrapper = await shallow(CvSearch, { propsData });

    expect(wrapper.vm.isToolbarKind).toBeTruthy();
  });

  it('`icon` is computed correctly when `size` is xl', async () => {
    const value = 'check-1';
    const propsData = { id: 'test-1', value, size: 'xl' };
    const wrapper = await shallow(CvSearch, { propsData });

    expect(wrapper.vm.icon).toBe(Search20);
  });

  it('`icon` is computed correctly when `size` is small', async () => {
    const value = 'check-1';
    const propsData = { id: 'test-1', value, size: 'small' };
    const wrapper = await shallow(CvSearch, { propsData });

    expect(wrapper.vm.icon).toBe(Search16);
  });

  it('`icon` is computed correctly when `size` is large', async () => {
    const value = 'check-1';
    const propsData = { id: 'test-1', value, size: 'large' };
    const wrapper = await shallow(CvSearch, { propsData });

    expect(wrapper.vm.icon).toBe(Search16);
  });

  it('`internalSize` is computed correctly when `size` is not set and `small` is true', async () => {
    const value = 'check-1';
    const propsData = { id: 'test-1', value, small: true };
    const wrapper = await shallow(CvSearch, { propsData });

    expect(wrapper.vm.internalSize).toBe('sm');
  });

  it('`internalSize` is computed correctly when `size` is not set and `small` is false', async () => {
    const value = 'check-1';
    const propsData = { id: 'test-1', value, small: false };
    const wrapper = await shallow(CvSearch, { propsData });

    expect(wrapper.vm.internalSize).toBe('xl');
  });

  it('`internalSize` is computed correctly when `size` is set and `small` is true', async () => {
    const value = 'check-1';
    const propsData = { id: 'test-1', value, small: true, size: 'large' };
    const wrapper = await shallow(CvSearch, { propsData });

    expect(wrapper.vm.internalSize).toBe('lg');
  });

  it('`internalSize` is computed correctly when `size` is set and `small` is false', async () => {
    const value = 'check-1';
    const propsData = { id: 'test-1', value, small: true, size: 'small' };
    const wrapper = await shallow(CvSearch, { propsData });

    expect(wrapper.vm.internalSize).toBe('sm');
  });

  it('`internalSize` is computed correctly when `size` is not correctly and `small` is false', async () => {
    const value = 'check-1';
    const propsData = { id: 'test-1', value, small: true, size: 'sdcdsc' };
    const wrapper = await shallow(CvSearch, { propsData });

    expect(wrapper.vm.internalSize).toBe('xl');
  });

  it('clear button click works as expected', async () => {
    const value = 'check-1';
    const propsData = { id: 'test-1', value };
    const wrapper = await shallow(CvSearch, { propsData });

    await trigger(wrapper.find('button'), 'click');

    expect(wrapper.vm.internalValue).toBe('');
    expect(wrapper.vm.clearVisible).toBeFalsy();
    expect(wrapper.emitted().input).toBeTruthy();
    expect(wrapper.emitted().input[0]).toEqual(['']);
  });

  it('on input event  works as expected', async () => {
    const value = 'check-1';
    const propsData = { id: 'test-1', value };
    const wrapper = await shallow(CvSearch, { propsData });

    const newValue = 'qwerty';
    wrapper.find('input').setValue(newValue);

    expect(wrapper.vm.clearVisible).toBeTruthy();
    expect(wrapper.vm.internalValue).toBe(newValue);
    expect(wrapper.emitted().input).toBeTruthy();
    expect(wrapper.emitted().input[0]).toEqual([newValue]);
  });

  it('`toggleActive` is called with `true` on toolbar button click', async () => {
    const value = 'check-1';
    const propsData = { id: 'test-1', value, kind: 'toolbar' };
    const wrapper = await shallow(CvSearch, { propsData });
    const spy = jest.spyOn(wrapper.vm, 'toggleActive');

    await trigger(wrapper.find('button'), 'click');

    expect(spy).toBeCalledWith(true);
  });
});
