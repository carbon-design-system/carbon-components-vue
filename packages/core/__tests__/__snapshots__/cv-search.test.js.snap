// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CvSearch should render correctly when \`clearAriaLabel\` is set 1`] = `
<cv-wrapper-stub tagtype="div" class="cv-search bx--form-item">
  <div role="search" class="bx--search bx--search--xl"><label for="test-1" class="bx--label"></label> <input id="test-1" type="text" role="search" placeholder="Search" aria-labelledby="test-1" class="bx--search-input">
    <!---->
    <anonymous-stub class="bx--search-magnifier"></anonymous-stub> <button type="button" title="test clear aria label" aria-label="test clear aria label" class="bx--search-close">
      <close16-stub></close16-stub>
    </button>
  </div>
</cv-wrapper-stub>
`;

exports[`CvSearch should render correctly when \`formItem\` is false 1`] = `
<cv-wrapper-stub tagtype="" class="cv-search bx--form-item">
  <div role="search" class="bx--search bx--search--xl cv-search"><label for="test-1" class="bx--label"></label> <input id="test-1" type="text" role="search" placeholder="Search" aria-labelledby="test-1" class="bx--search-input">
    <!---->
    <anonymous-stub class="bx--search-magnifier"></anonymous-stub> <button type="button" title="Clear search input" aria-label="Clear search input" class="bx--search-close">
      <close16-stub></close16-stub>
    </button>
  </div>
</cv-wrapper-stub>
`;

exports[`CvSearch should render correctly when \`kind\` is toolbar 1`] = `
<cv-wrapper-stub tagtype="div" class="cv-search bx--form-item">
  <div role="search" class="bx--search bx--search--xl bx--toolbar-search"><label for="test-1" class="bx--label"></label> <input id="test-1" type="text" role="search" placeholder="Search" aria-labelledby="test-1" class="bx--search-input"> <button type="button" aria-label="Toolbar search" class="bx--toolbar-search__btn">
      <anonymous-stub class="bx--search-magnifier"></anonymous-stub>
    </button>
    <!----> <button type="button" title="Clear search input" aria-label="Clear search input" class="bx--search-close">
      <close16-stub></close16-stub>
    </button></div>
</cv-wrapper-stub>
`;

exports[`CvSearch should render correctly when \`placeholder\` is set 1`] = `
<cv-wrapper-stub tagtype="div" class="cv-search bx--form-item">
  <div role="search" class="bx--search bx--search--xl"><label for="test-1" class="bx--label"></label> <input id="test-1" type="text" role="search" placeholder="test placeholder" aria-labelledby="test-1" class="bx--search-input">
    <!---->
    <anonymous-stub class="bx--search-magnifier"></anonymous-stub> <button type="button" title="Clear search input" aria-label="Clear search input" class="bx--search-close">
      <close16-stub></close16-stub>
    </button>
  </div>
</cv-wrapper-stub>
`;

exports[`CvSearch should render correctly when \`size\` is large 1`] = `
<cv-wrapper-stub tagtype="div" class="cv-search bx--form-item">
  <div role="search" class="bx--search bx--search--lg"><label for="test-1" class="bx--label"></label> <input id="test-1" type="text" role="search" placeholder="Search" aria-labelledby="test-1" class="bx--search-input">
    <!---->
    <anonymous-stub class="bx--search-magnifier"></anonymous-stub> <button type="button" title="Clear search input" aria-label="Clear search input" class="bx--search-close">
      <close16-stub></close16-stub>
    </button>
  </div>
</cv-wrapper-stub>
`;

exports[`CvSearch should render correctly when \`size\` is small 1`] = `
<cv-wrapper-stub tagtype="div" class="cv-search bx--form-item">
  <div role="search" class="bx--search bx--search--sm"><label for="test-1" class="bx--label"></label> <input id="test-1" type="text" role="search" placeholder="Search" aria-labelledby="test-1" class="bx--search-input">
    <!---->
    <anonymous-stub class="bx--search-magnifier"></anonymous-stub> <button type="button" title="Clear search input" aria-label="Clear search input" class="bx--search-close">
      <close16-stub></close16-stub>
    </button>
  </div>
</cv-wrapper-stub>
`;

exports[`CvSearch should render correctly when \`size\` is xl 1`] = `
<cv-wrapper-stub tagtype="div" class="cv-search bx--form-item">
  <div role="search" class="bx--search bx--search--xl"><label for="test-1" class="bx--label"></label> <input id="test-1" type="text" role="search" placeholder="Search" aria-labelledby="test-1" class="bx--search-input">
    <!---->
    <anonymous-stub class="bx--search-magnifier"></anonymous-stub> <button type="button" title="Clear search input" aria-label="Clear search input" class="bx--search-close">
      <close16-stub></close16-stub>
    </button>
  </div>
</cv-wrapper-stub>
`;

exports[`CvSearch should render correctly when \`toolbarAriaLabel\` is set 1`] = `
<cv-wrapper-stub tagtype="div" class="cv-search bx--form-item">
  <div role="search" class="bx--search bx--search--xl bx--toolbar-search"><label for="test-1" class="bx--label"></label> <input id="test-1" type="text" role="search" placeholder="Search" aria-labelledby="test-1" class="bx--search-input"> <button type="button" aria-label="test toolbar aria label" class="bx--toolbar-search__btn">
      <anonymous-stub class="bx--search-magnifier"></anonymous-stub>
    </button>
    <!----> <button type="button" title="Clear search input" aria-label="Clear search input" class="bx--search-close">
      <close16-stub></close16-stub>
    </button></div>
</cv-wrapper-stub>
`;

exports[`CvSearch should render correctly with only \`value\` prop set 1`] = `
<div class="cv-search bx--form-item">
  <div role="search" class="bx--search bx--search--xl"><label for="test-1" class="bx--label"></label> <input id="test-1" type="text" role="search" placeholder="Search" aria-labelledby="test-1" class="bx--search-input">
    <!----> <svg focusable="false" preserveAspectRatio="xMidYMid meet" xmlns="http://www.w3.org/2000/svg" fill="currentColor" width="16" height="16" viewBox="0 0 16 16" aria-hidden="true" class="bx--search-magnifier">
      <path d="M15,14.3L10.7,10c1.9-2.3,1.6-5.8-0.7-7.7S4.2,0.7,2.3,3S0.7,8.8,3,10.7c2,1.7,5,1.7,7,0l4.3,4.3L15,14.3z M2,6.5	C2,4,4,2,6.5,2S11,4,11,6.5S9,11,6.5,11S2,9,2,6.5z"></path>
    </svg> <button type="button" title="Clear search input" aria-label="Clear search input" class="bx--search-close"><svg focusable="false" preserveAspectRatio="xMidYMid meet" xmlns="http://www.w3.org/2000/svg" fill="currentColor" width="16" height="16" viewBox="0 0 32 32" aria-hidden="true">
        <path d="M24 9.4L22.6 8 16 14.6 9.4 8 8 9.4 14.6 16 8 22.6 9.4 24 16 17.4 22.6 24 24 22.6 17.4 16 24 9.4z"></path>
      </svg></button></div>
</div>
`;
